proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,270 @@
+contains two integers (for example, an indentification number and an index into
+a street address table).
+
+A given pair of integers might be either a `Chair` or a `Person` (or a 2D point)
+and there is no way of conveying the distinction unless the nominal type is also
+taken into consideration.
+
+### Algebraic Type Definitions
+
+The Interface Types language permits the definition of a limited form of
+_algebraic type definition_. This is, in part, to regularize concepts such as
+nullability and to permit limited forms of polymorphism (allowing a color to be
+named by a string, a vector of small integers or a large integer, for example).
+
+WebIDL explicitly prohibits the formation of new types; indeed it uses type
+unions to achieve similar polymorphism. However, as we note above, not all
+programming languages support dynamic types or type unions; which makes it
+problematic to support such polymorphism.
+
+The form of algebraic types envisaged in the Interface Types system is
+intentionally limited. For example, we do not permit _recursive_ types to be
+defined. This is in keeping with the vision of supporting APIs across ownership
+boundaries where complex data is likely to take limited form.[^This design
+choice may be revisited in future versions of the Interface Types specification
+if supporting structured types such as abstract XML and/or JSON become
+important.]
+
+### Working with WebIDL
+
+While the Interface Types language introduces its own IDL, it is still clearly
+important to support WebIDL; particularly for Web APIs.
+
+The strategy that we adopt can be summarized as interpreting WebIDL as an
+alternate target to WebAssembly itself.
+
+When using Interface Types to specify a public API for a WebAssembly module, the
+author must provide special _adapters_ -- import adapters for accessing external
+APIs and export adapters when offering a public API.
+
+Typically, such adapters would be automatically generated by language
+specific tools; which would mean that individual code authors would likely be
+not involved in the construction of the adapters.
+
+Similarly, in the case where a WebIDL API is involved, there would be adapters
+between the Interface Types language and WebIDL. There would also be adapters
+between WebIDL and the host environment; although these are usually private to
+the host.
+
+In case case of adapting between WebIDL and Interface Types, it is hoped that a
+single method for automatically constructing adapters would be sufficient; and
+automatic. This is should be possible, based on the fact that the Interface
+Types language has been designed to be a slight extension of a simplification of
+WebIDL.
+
+Note that this does not imply that the semantics of Interface Types and WebIDL
+are completely aligned: any automatically generated adapters would likely ignore
+those aspects of WebIDL that are not accounted for in the Interface Types
+language.
```

Comment:
I don't think we should have to bring up the concept of adapters to explain how Web IDL and IT will interoperate.  I think once you frame IT *not* as an IDL, but as just introducing a new set of types and values of these new types, we can just say that, just as calling a Web IDL function from JS is governed by the [Web IDL ECMAScript Binding](https://heycam.github.io/webidl/#ecmascript-binding), which takes a set of JS values and says how to create Web IDL values out of them, we will add a new "Web IDL WebAssembly Binding" that takes a set of interface values and how to create Web IDL values of them.  Again, I think symmetry with JS is key.

Then we need to actually go down the list of Web IDL types and give a sense of what we'd specify; people will particularly want to know about strings.

Reply:
Might work. Except that it would be the other way around: take the WebIDL values and show how they can be realized from Interface Types values.